### PR TITLE
Add Docker to release

### PR DESCRIPTION
**Pull request by swetharepakula**
- bump blockhead
- add eth.json to blockhead job
- increase disk of blockhead job

Signed-off-by: Swetha Repakula <srepaku@us.ibm.com>

### CODE REVIEW

**Review by MHBauer — APPROVED** [review]

**Comment by nimakaviani** [comment]
the broker deploys fine and querying the service catalog returns the service. but I get the following error when trying to provision the service:

```
{"description":"Error response from daemon: cgroups: cgroup deleted: unknown"}
```

**Comment by nimakaviani** [comment]
We might be able to merge this and create another issue for the service provisioning problem

File: jobs/blockhead/spec [hunk 1]
```diff
@@ -15,3 +16,5 @@ properties:
     description: "Username for Blockhead broker"
   password:
     description: "Password for Blockhead broker"
+  port:
+    description: "Port for Blockhead broker to run on"
```
Comment (nimakaviani) [inline jobs/blockhead/spec:19]: can we set a default `3333` for the port in the spec and get rid of the port property in the manifest.

File: jobs/blockhead/templates/eth.json.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "plans": [
+        {
+            "name": "geth-dev",
+            "free": true,
+            "image": "nimak/eth-dev:latest",
+            "description": "Free Service that deploys a geth node in development mode",
+            "ports": ["8545"]
+        }
+    ]
+}
```
Comment (nimakaviani) [inline jobs/blockhead/templates/eth.json.erb]: embedding the service config into the job template is a bit counter intuitive. This basically means that for every service that a user wants to add to the release, they need to also modify the spec and rename the file to be a `json.erb` file.

Can we instead fix this during packaging, for example by copying everything under `src/github.com/cloudfoundry-incubator/blockhead/services` (or alternatively a dedicated `services/` directory in the release) into `${BOSH_INSTALL_TARGET}/services/` and referencing that in the bpm script.

**Comment by swetharepakula** [comment]
Oh that is interesting. We didn't see that error on our side. After deploying you should be able to provision and deprovision without an error. The error looks like it is coming from docker?

File: jobs/blockhead/spec (hunk 1 above)
Comment (swetharepakula) [reply jobs/blockhead/spec:19]: I was going back and forth on this one. Blockhead already defaults the port. I am thinking we should make it required in blockhead and default here in the release, but I will make that change.

File: jobs/blockhead/templates/eth.json.erb (hunk 2 above)
Comment (swetharepakula) [reply jobs/blockhead/templates/eth.json.erb]: That seems a little weird to me. I didn't think services should be part of the blockhead source code package. Other thing I don't like about that approach is that we are tying the release to sample files in blockhead. I would think the bosh release should be where the source of truth of services not the blockhead source code.

File: jobs/blockhead/spec (hunk 1 above)
Comment (nimakaviani) [reply jobs/blockhead/spec:19]: yeah that makes sense

**Comment by nimakaviani** [comment]
yeah looks like a docker problem. This is the docker that comes from the release. I wonder if there is issues there. Did you actually try to provision and deprovision after you deployed?

File: jobs/blockhead/templates/eth.json.erb (hunk 2 above)
Comment (nimakaviani) [reply jobs/blockhead/templates/eth.json.erb]: I agree. That's why I suggested maybe we can have a `services/` directory in the bosh release and we move service files from there into the deployment. The directory doesn't need to have anything in it to begin with, but if created, we can copy arbitrary set of services files through to the deployment. Let's discuss it further in the meeting.

**Comment by swetharepakula** [comment]
Yup, both of them work. We monitored the docker images being pulled and the containers created and destroyed. Must mean there is some difference that didn't get caught in the release.

**Comment by nimakaviani** [comment]
Here is the logs from the docker job

```
time="2018-09-20T02:40:12.809052166Z" level=error msg="Handler for POST /v1.35/containers/e4779a1dae9e330f5231fa88a6c7ba4f3a1f527ddc7938de1f55b2a965ff03d2/start returned error: cgroups: cgroup deleted: unknown"
time="2018-09-20T02:41:29Z" level=info msg="shim docker-containerd-shim started" address="/containerd-shim/moby/1d87cb1665513c9e3d029fcb95ba1c8833838794aa414b67f77fd70eefd28d3f/shim.sock" debug=false module="containerd/tasks" pid=16353
time="2018-09-20T02:41:29Z" level=error msg="failed to disable IPv6 forwarding for container's interface all: open /proc/sys/net/ipv6/conf/all/disable_ipv6: no such file or directory"
time="2018-09-20T02:41:29.397098478Z" level=warning msg="Failed to disable IPv6 on all interfaces on network namespace \"/var/run/docker/netns/1c5d2b91ef6a\": reexec to set IPv6 failed: exit status 4"
time="2018-09-20T02:41:29.565082387Z" level=error msg="stream copy error: read /proc/self/fd/66: file already closed"
time="2018-09-20T02:41:29.581390850Z" level=error msg="failed to delete task after fail start" container=1d87cb1665513c9e3d029fcb95ba1c8833838794aa414b67f77fd70eefd28d3f error="task must be stopped before deletion: running: failed precondition" module=libcontainerd namespace=moby
time="2018-09-20T02:41:29.597960597Z" level=error msg="failed to delete failed start container" container=1d87cb1665513c9e3d029fcb95ba1c8833838794aa414b67f77fd70eefd28d3f error="cannot delete running task 1d87cb1665513c9e3d029fcb95ba1c8833838794aa414b67f77fd70eefd28d3f: failed precondition"
time="2018-09-20T02:41:29.880783106Z" level=error msg="1d87cb1665513c9e3d029fcb95ba1c8833838794aa414b67f77fd70eefd28d3f cleanup: failed to delete container from containerd: cannot delete running task 1d87cb1665513c9e3d029fcb95ba1c8833838794aa414b67f77fd70eefd28d3f: failed precondition"
time="2018-09-20T02:41:29.880886446Z" level=error msg="Handler for POST /v1.35/containers/1d87cb1665513c9e3d029fcb95ba1c8833838794aa414b67f77fd70eefd28d3f/start returned error: cgroups: cgroup deleted: unknown"
```

**Comment by nimakaviani** [comment]
the modified version looks good to me:

setup:

```
bosh-deployment sha: ba03a10581182bbf29efa586040f16064ab92f7d
stemcell xenial v97.18: Full Stemcell – sha1:3cb930e01ac4d8612a255ea1171185e595b09194
```

when deploying bosh-deployment with the above `sha`, it complained about the wrong release sha for grootfs. had to make the following change for it to download the compiled release successfully:

```
-    sha1: fa426dd6545e101471c3dc1124b70cc336c101c
+    sha1: 7fa426dd6545e101471c3dc1124b70cc336c101c
```

with those steps done, the deploy was successful. 

first time provisioning of the service takes a lot of time to pull the image. makes me wonder whether we should try and pull the service image in the docker repository for the list of servieces as the deploy goes through? 

- [x] deploy works
- [x] default broker port is set in the manifest
- [x] services directory is under `src/services`

**Pull request merged**